### ISSUE TITLE
Add a mana usage "rating"

### ISSUE THREAD

**Issue by MartijnHols** [WoWAnalyzer/WoWAnalyzer#164]
Suggestions like `Your IoL FoL to HL cast ratio can likely be improved. When you get an Infusion of Light proc try to cast Flash of Light as much as possible, it is a considerably stronger heal (20 Flash of Lights.` are wrong when a user has a mana graph like this:
![image](https://user-images.githubusercontent.com/4565223/29438249-b87a35c0-83b5-11e7-8c78-1464382f4763.png)
Relevant log: https://wowanalyzer.com/report/TDFkBHmfAbjrwCz3/8-Mythic+Fallen+Avatar+-+Kill+(7:42)/Ezy/mana

This suggestion could be disabled if ending mana is below a threshold, but that would be a clunky way of doing it. Instead maybe ManaValues could give the mana usage a rating of one of POOR, DECENT, GOOD so that this simplified metric can be used instead (e.g. GOOD = end with or lowest at <40,000 mana, DECENT <100,000 mana, else POOR).

The above metric would be completely disabled if GOOD, limited to minor at DECENT and fully enabled at POOR. Or percentages could shift around. idk

Also could add suggestions based on the rating.

It's basically a simplified mana metric that takes care of the proper thresholds making it much easier to reason about.

**Comment by ronaldpereira** [comment]
I thought of a more complex, but more certain mana metric for that.

The goal of a healer in a raid encounter is to always have the mana % equal to the boss %. So if in each fight moment we get the
`
const manaDifferences = 0;
manaDifferences += abs(currentManaPercentage-bossHealthPercentage);
`
and after getting this value, we just have to do
`
const manaMetric = manaDifferences / this.owner.fightDuration;
`
in order to get the average mana difference between healer mana and boss health.

And then we can set a parameter for that manaMetric as:
`
Perfect: manaMetric <= 0.05
Good: manaMetric <= 0.10
Decent: manaMetric <= 0.15
Poor: manaMetric > 0.15
`
What do you think of that?

**Comment by MartijnHols** [comment]
`The goal of a healer in a raid encounter is to always have the mana % equal to the boss %.` that's just a guideline, in the end the only thing that matters is if you ever OOMed.

**Comment by ronaldpereira** [comment]
I think with this manaMetric I suggested, if you ever OOMed, it would show this as well. But imagine: a healer have used everything in the beginning of the encounter and stayed with 15% mana cap all fight and then in the last 5% of boss health, he dropped to 2% mana. That metric you suggested would say that his rating is good, which isn't.

I think having a metric that gets all the fight it's better for an analysis. The goal of an analyzer is to get something that is imperceptible and output them in a organized and clear way.

I don't have the expertise in healing specs, but I imagine this should be a great way to measure healer performances in a metric way.

**Comment by MartijnHols** [comment]
I feel a player's performance rating and spell usage and other suggestions would suffer if that were the case, which should reveal that that playstyle too isn't efficient because other things start to suffer.

But I agree with you that most of the time such a metric could more accurately reveal mana usage quality, but I think it would be wrong on a bunch of fights where you either need to spend more mana early or late in the fight and you should deviate from the line. As I said earlier, following the boss health line is merely an guideline.

I feel that such an approach would over-complicate the issue without a great enough gain to warrant it. This has two issues: obviously programmatically where both maintenance and usage become a little harder, but also more importantly for users. As a user would expect a mana performance rating to be based around whether you ran OOM during the fight as that's usually the only thing that matters. If we use a differently approach it could be confusing to them and we'd probably have to explain it. This would be hard to do in such a way that users actually read and understand it, and would be additional information clutter.

To summarize; my personal opinion is that it would over complicate the metric without real gain. Users playing inefficiently with their mana that wouldn't be spotted by my suggestion approach should notice a negative effect on other metrics.

**Comment by ronaldpereira** [comment]
I agree with you. I that case, the analysis should be way more simple than I've imagined.
